api/firmware/device.go

```diff
@@ -241,11 +253,25 @@ func (device *Device) Init() error {
 	device.channelHashDeviceVerified = false
 	device.sendCipher = nil
 	device.receiveCipher = nil
-	device.changeStatus(StatusConnected)
 
-	if err := device.inferVersionAndProduct(); err != nil {
-		return err
+	version, product, unlocked, initialized, err := device.info()
+	if err != nil {
+		return errp.New(
+			"OP_INFO unavailable; need to provide version and product via the USB HID descriptor")
```

Comment:
This is incorrect - old firmwares don't support OP_INFO and we can't just error here. Before, the OP_INFO call to infer the version/product would be done only if it was not already provided.

What you now want to do is probably to call info() if supported by the firmware (need to look up fw version), and use the results of it when possible.

Reply:
Ah yes. We do need to call `info()` if we don't know if its supported `device.version == nil` too tough, so I'll just put this into a `version == nil || version atLeast(4.3.0)`

Reply:
As mentioned above here:
https://github.com/BitBoxSwiss/bitbox02-api-go/pull/98#discussion_r1785303280

I think changing the if-clause around this to:
`if device.version == nil || device.version.AtLeast(semver.NewSemVer(9, 2, 0)) { ...}`
is the way to go.

_side note_: But I also think that it was not incorrect though, we basically do the same already:
https://github.com/BitBoxSwiss/bitbox02-api-go/blob/a2115fee878e796dce1edcc670876b734fef282b/api/firmware/device.go#L247

Just propagated from the `inferVersionAndProduct` function instead of directly.